packages/neon/neon/lib/src/pages/home.dart

```diff
@@ -106,39 +106,26 @@ class _HomePageState extends State<HomePage> {
               const drawer = NeonDrawer();
               const appBar = NeonAppBar();
 
-              Widget body = Builder(
-                builder: (final context) => Column(
-                  children: [
-                    if (appImplementations.data != null) ...[
-                      if (appImplementations.data!.isEmpty) ...[
-                        Expanded(
-                          child: Center(
-                            child: Text(
-                              AppLocalizations.of(context).errorNoCompatibleNextcloudAppsFound,
-                              textAlign: TextAlign.center,
-                            ),
-                          ),
-                        ),
-                      ] else ...[
-                        if (activeAppIDSnapshot.hasData) ...[
-                          Expanded(
-                            child: appImplementations.data!.find(activeAppIDSnapshot.data!)!.page,
-                          ),
-                        ],
-                      ],
-                    ],
-                  ],
-                ),
-              );
+              final appView = appImplementations.data?.find(activeAppIDSnapshot.data ?? '')?.page;
+
+              Widget? noAppsView;
+              if (appImplementations.data?.isEmpty ?? false) {
+                noAppsView = Center(
+                  child: Text(
+                    AppLocalizations.of(context).errorNoCompatibleNextcloudAppsFound,
+                    textAlign: TextAlign.center,
+                  ),
+                );
+              }
 
-              body = MultiProvider(
+              Widget body = MultiProvider(
                 providers: _appsBloc.appBlocProviders,
                 child: Scaffold(
                   key: _scaffoldKey,
                   resizeToAvoidBottomInset: false,
                   drawer: !drawerAlwaysVisible ? drawer : null,
                   appBar: appBar,
-                  body: body,
+                  body: appView ?? noAppsView,
```

Comment:
I would rather use the null fall through operator to construct the no apps view here. Otherwise you are checking two conditions and they are not even the same (which could lead to problems for edge-cases).

Reply:
Do you mean like:
```dart
Scaffold(
  body: appView ?? Center(child: Text('no apps')),
);
```
?

This does not work because when the appimplementations haven't finished loading yet they are null wich would result in the 'no apps found error' being shown while loading the app.


I think I`ll revert it to the builder and return a empty SizedBox when loading. At least removing all the collum stuff:

Something like:
```dart
final body = Builder(builder: (context){
  if(appimplementations == null){
    return const SizedBox();
  }
  
  if(appimplementations.isEmpty){
    return // error page;
  }
  
  return appimplementations.currentPage;
});
```

Reply:
Yes, but that would only be shown on the very first load, as the cached data would be shown otherwise.

Reply:
Yeah but I'd rather show an empty container (or SizedBox) than the error message.

Also depending on timing the error message would only be shown for a very brief time wich wouldn't be good for a11y